### PR TITLE
Make tsc faster again

### PR DESCRIPTION
**Pull request by t3chguy**
With some profiling I found a hotspot around AccessibleButton's types so rewrote them to make them to be faster and more correct. First commit focuses on speed, the rest on correctness. It found a bunch of props being given to divs which don't make sense without swapping to `button`.

Before 6m 51s
After 57s

### CODE REVIEW

File: src/components/views/dialogs/devtools/SettingExplorer.tsx [hunk 1]
```diff
@@ -298,7 +298,7 @@ const SettingsList: React.FC<ISettingsListProps> = ({ onBack, onView, onEdit })
                                     <code>{i}</code>
                                 </AccessibleButton>
                                 <AccessibleButton
-                                    alt={_t("devtools|edit_setting")}
+                                    title={_t("devtools|edit_setting")}
```
Comment (dbkr) [inline src/components/views/dialogs/devtools/SettingExplorer.tsx:301]: There are a few of these alt/title changes - are they related?

Comment (t3chguy) [reply src/components/views/dialogs/devtools/SettingExplorer.tsx:301]: Yes, `alt` is not a valid property on a div.

Comment (dbkr) [reply src/components/views/dialogs/devtools/SettingExplorer.tsx:301]: 👍

**Review by dbkr — APPROVED** [review]
Thanks, looking forward to this being usable again!

**Pull request merged**